src/BatteryStateOfChargeService.h

```diff
@@ -42,8 +42,13 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     void addData(const BatteryData& batteryData);
 
 private:
-    // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
-
-    // Recommend adding necessary variables here
+    double newCurrent_;
+    double totalCurrent_;
+    QTime estimatedTime_;
+    double oldCurrent_;
+    QTime oldTime_ = QTime(0,0,0);
```

Comment:
Private members type custom objects (not primitive types) will often be initialied with their default constructor. In this case `oldTime_` would be constructed with `QTime()` which would effectively be the same as `QTime(0, 0, 0)`.